plugins/modules/ndo_qos_class_policy.py

```diff
@@ -0,0 +1,498 @@
+          minimum_threshold: 5
+          maximum_threshold: 95
+          probability: 80
+          weight: 1
+        scheduling_algorithm: weighted_round_robin
+        bandwidth_allocated: 50
+        pfc_admin_state: enabled
+        admin_state: enabled
+        no_drop_cos: cos1
+        pfc_scope: intra_tor
+    state: present
+  register: add_qos_class_policy_level2
+
+- name: Update a QoS Class policy by removing QoS level2 and keeping QoS level1
+  cisco.mso.ndo_qos_class_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    name: ansible_test_qos_class_policy
+    description: Ansible Test QoS Class Policy
+    preserve_cos: true
+    qos_levels:
+      - level: level1
+    state: present
+  register: remove_qos_class_policy_level2
+
+- name: Update a QoS Class policy by removing all QoS levels
+  cisco.mso.ndo_qos_class_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    name: ansible_test_qos_class_policy
+    description: Ansible Test QoS Class Policy
+    preserve_cos: true
+    qos_levels: []
+    state: present
+  register: remove_qos_class_policy_all_levels
+
+- name: Query QoS Class policy using name
+  cisco.mso.ndo_qos_class_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    name: ansible_test_qos_class_policy
+    state: query
+  register: query_one
+
+- name: Query QoS Class policy using UUID
+  cisco.mso.ndo_qos_class_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    uuid: "{{ create_qos_class_policy.current.uuid }}"
+    state: query
+  register: query_one_uuid
+
+- name: Delete a QoS Class policy using name
+  cisco.mso.ndo_qos_class_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    name: ansible_test_qos_class_policy
+    state: absent
+
+- name: Delete a QoS Class policy using UUID
+  cisco.mso.ndo_qos_class_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    uuid: "{{ query_one.current.uuid }}"
+    state: absent
+"""
+
+RETURN = r"""
+"""
+import copy
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from ansible_collections.cisco.mso.plugins.module_utils.template import MSOTemplate, KVPair
+from ansible_collections.cisco.mso.plugins.module_utils.utils import append_update_ops_data, delete_none_values
+from ansible_collections.cisco.mso.plugins.module_utils.constants import (
+    QOS_CONGESTION_ALGORITHM_MAP,
+    QOS_SCHEDULING_ALGORITHM_MAP,
+    QOS_PFC_SCOPE_MAP,
+    QOS_LEVEL,
+)
+
+
+def main():
+    qos_level_list = copy.copy(QOS_LEVEL)
```

Comment:
should we name the variable better to describe its use case and put some inline comment why the unspecified is removed